README.md

```diff
@@ -1 +1,304 @@
+port = 2345
+database = "cachedb"
+username = "admin"
+password = "secret"
+```
+
+- Matcher.sync_interval: Minimal delay between matching cycles.
+- Matcher.match_threshold: Value for showing possible matches
+- Matcher.asset_plugins_path: Path to the directory containing asset-specific plugin configuration files.
+- Matcher.csaf_plugins_path: Path to the directory containing CSAF-specific plugin configuration files.
+- Matcher.Api.host/port: Address where the FastAPI server listens.
+- Matcher.Cachedb: Connection to the shared cache DB used by all components.
+
+#### Asset Synchronizer configuration (assets/assetsync.toml)
+
+Example (defaults in repo):
+
+```toml
+[Assetsync]
+# Asset-specific options go here (plugin-specific)
+
+[Synchronizer]
+sync_interval = 60
+preprocessor_plugins = ["identity"]
+plugin_configs_path = "./assets/plugin_configs/data_source/asset"
+cleanup_grace_period = 3600
+
+[Synchronizer.Api]
+host = "0.0.0.0"
+port = 8992
+
+[Cachedb]
+host = "localhost"
+port = 2345
+database = "cachedb"
+username = "admin"
+password = "secret"
+```
+
+- Synchronizer.sync_interval: Minimal delay between fetch cycles.
+- Synchronizer.preprocessor_plugins: List and order of preprocessing plugins.
+- Synchronizer.plugin_configs_path: Path to the directory containing plugin configuration files.
+- Synchronizer.cleanup_grace_period: Time in seconds from the last synchronization after which assets are considered stale and will be deleted.
+- Synchronizer.Api.host/port: Address for the synchronizer API.
+- Cachedb: Connection to the shared cache DB.
+- Data source plugins: Configuration TOML files are loaded from assets/plugin_configs/asset_source/…
+
+#### CSAF Synchronizer configuration (assets/csafsync.toml)
+
+Example (defaults in repo):
+
+```toml
+[Csafsync]
+# CSAF-specific options go here (plugin-specific)
+
+[Synchronizer]
+sync_interval = 60
+preprocessor_plugins = ["identity"]
+plugin_configs_path = "./assets/plugin_configs/data_source/csaf"
+cleanup_grace_period = 3600
+
+[Synchronizer.Api]
+host = "0.0.0.0"
+port = 8991
+
+[Cachedb]
+host = "localhost"
+port = 2345
+database = "cachedb"
+username = "admin"
+password = "secret"
+```
+
+- Same meaning as for the Asset Synchronizer
+
+## Authenticate with the API
+
+The synchronizer components (Asset/CSAF) and the Matcher expose a small FastAPI HTTP API that uses OAuth2 Password flow to issue short‑lived JWT bearer tokens.
+
+Quick start:
+
+1) Make sure the component is running (e.g., Matcher default at <http://localhost:8998/docs>; configurable in `assets/matcher.toml`) after executing the setup and configuration.
+2) Create or update a user in the CacheDB using the CLI:
+
+```bash
+uv run csaf_matcher_cli user create -u admin
+```
+
+### Local Setup
+
+When using the local setup, the password has to be stored in `\dev\configuration\plugins.py`
+
+```python
+    'synchronisers': {
+        'username': 'admin',
+        'password': '<password>',
+        ...
+    }
+```
```

Comment:
Is that still needed? It feels weird to have a password in the code (that might be committed to the repo)...
But maybe, I'm missing something.

Reply:
Providing the password in the plugin configuration file is necessary, but i would prefer to change this such that it is provided to the docker installation via an environment variable and then automatically placed inside this config file. That way nothing will get commited accidentally. I will look into this.